src/library/scala/collection/mutable/TreeSet.scala

```diff
@@ -29,88 +28,162 @@ object TreeSet extends MutableSortedSetFactory[TreeSet] {
    */
   def empty[A](implicit ordering: Ordering[A]) = new TreeSet[A]()
 
+  /** $sortedMapCanBuildFromInfo */
+  implicit def canBuildFrom[A](implicit ord: Ordering[A]): CanBuildFrom[Coll, A, TreeSet[A]] =
+    new SortedSetCanBuildFrom[A]
 }
 
 /**
- * A mutable SortedSet using an immutable RedBlack Tree as underlying data structure.
+ * A mutable sorted set implemented using a mutable red-black tree as underlying data structure.
  *
- * @author Lucien Pereira
+ * @param ordering the implicit ordering used to compare objects of type `A`.
+ * @tparam A the type of the keys contained in this tree set.
+ *
+ * @author Rui Gonçalves
+ * @version 2.12
+ * @since 2.10
  *
+ * @define Coll mutable.TreeSet
+ * @define coll mutable tree set
  */
 @deprecatedInheritance("TreeSet is not designed to enable meaningful subclassing.", "2.11.0")
```

Comment:
I'm not sure there's any way that previous inheritance would possibly remain intact.  You might just make it final and see what breaks.  If things do break, they will probably break less badly than they might at runtime due to assumptions being violated.